### ISSUE TITLE
There is a little mistake in your unit value

### ISSUE THREAD

**Issue by syedsimanta03** [stylelint/stylelint#3932]
html,
body {
  font-family: serif;
  line-height: 1.7;
}

I didn't use any unit em or px and it's not showing any error.

**Comment by jeddy3** [comment]
If you like, you can enforce the use of the `em` unit for the `line-height` property using the [`declaration-property-value-whitelist`](https://github.com/stylelint/stylelint/tree/master/lib/rules/declaration-property-value-whitelist) rule, like so:

```json
{
  "rules": {
    "declaration-property-value-whitelist": {
      "line-height": ["/.*em$/"]
    }
  }
}
```

You can tweak the regex to enforce other units.

**Comment by syedsimanta03** [comment]
Please let me know if there is any rule which can ditect missing unit like we have to use unit after number or padding marging etc. If not this will show error
--
Sent from Mail.Ru app for Android Tuesday, 29 January 2019, 00:28a.m. +06:00 from jeddy3  notifications@github.com :

**Comment by jeddy3** [comment]
> Please let me know if there is any rule which can ditect missing unit like we have to use unit after number or padding marging etc.

For `padding` and `margin` you can use the [`stylelint-csstree-validator`](https://www.npmjs.com/package/stylelint-csstree-validator) plugin. This _validates your property value pairs against the CSS specification_.

Alternatively, you can track https://github.com/stylelint/stylelint/issues/3208#issuecomment-453754765 which is a proposal to build a similar plugin but it intends to support more non-standard syntax like SCSS and Less.

**Comment by syedsimanta03** [comment]
Thanks. You are the star! Is there any rule that when I am writing class and id names in a css style file, if I write a wrong id class name which doesn't match html id class stylelint will show me an error?

**Comment by jeddy3** [comment]
> Is there any rule that when I am writing class and id names in a css style file

No, there isn't. stylelint conducts static analysis on your styles, and so the built-in rules are unaware of the structure of your HTML. However, you could [write a bespoke plugin](https://stylelint.io/developer-guide/plugins/) that parses your HTML files, extracts the ids and classes from it, and then checks the id and class selectors within your styles.

**Comment by syedsimanta03** [comment]
when I use line-height: 1.2; without any unit like em, rem it shows an error that there is no unit or whitelist. So there is no rule I found to fix, is there any rule -only line height can be without unit but other properties must have the unit.

**Comment by syedsimanta03** [comment]
Hi please check the image and I want to keep the custom variables like that way, but itz show declaration-empty-line-before error. Can you please set a rules. I don't need any empty line rule here.

**Comment by syedsimanta03** [comment]
what should I do the unitless line-height: 1.2; shows error. I want to use unitless line-height: 1.2;
